### PR TITLE
fix on FreeBSD

### PR DESCRIPTION
**Pull request by swills**
Fixes build and test on FreeBSD 11 with Go 1.5 for me.

### CODE REVIEW

File: termios/pty_bsd.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package termios
```
Comment (davecheney) [inline termios/pty_bsd.go]: You need to add
```
// +build darwin freebsd openbsd netbsd

```
Above the `package` declaration to scope this file to bsd like things (naming the file _bsd.go doesn't do anything by itself)

File: term_bsd.go [hunk 2]
```diff
@@ -7,7 +7,7 @@ import "syscall"
 type attr syscall.Termios
 
 func (a *attr) setSpeed(baud int) error {
-	var rate uint64
+	var rate uint32
```
Comment (davecheney) [inline term_bsd.go]: Why did you change this ? If you need to make a change for only freebsd, you'll need to copy this file into term_freebsd.go and alter the build tags on the other files.

File: termios/termios_bsd.go [hunk 3]
```diff
@@ -73,10 +73,10 @@ func Tcflush(fd, which uintptr) error {
 }
 
 // Cfgetispeed returns the input baud rate stored in the termios structure.
-func Cfgetispeed(attr *syscall.Termios) uint64 { return attr.Ispeed }
+func Cfgetispeed(attr *syscall.Termios) uint32 { return attr.Ispeed }
```
Comment (davecheney) [inline termios/termios_bsd.go]: Same as above

**Comment by davecheney** [comment]
Thanks for this. I have some comments and questions, please let me know when you're ready for the next round of review.

**Comment by davecheney** [comment]
Thanks for this. I have some comments and questions, please let me know when you're ready for the next round of review.

File: term_bsd.go (hunk 2 above)
Comment (swills) [inline term_bsd.go]: Without this, I get:

swills/term/term_bsd.go:51: cannot use syscall.CS8 | syscall.CREAD | syscall.CLOCAL | rate (type uint64) as type uint32 in assignment
swills/term/term_bsd.go:52: cannot use rate (type uint64) as type uint32 in assignment
swills/term/term_bsd.go:53: cannot use rate (type uint64) as type uint32 in assignment
FAIL    github.com/swills/term [build failed]

Don't know if this is needed on other OSs, but can update if needed.

File: termios/termios_bsd.go (hunk 3 above)
Comment (swills) [inline termios/termios_bsd.go]: This was because of a build error:

term/termios/termios_bsd.go:76: cannot use attr.Ispeed (type uint32) as type uint64 in return argument
term/termios/termios_bsd.go:79: cannot use attr.Ospeed (type uint32) as type uint64 in return argument

The docs on Termios made me think uint32 was right:

https://golang.org/pkg/syscall/#Termios

**Comment by swills** [comment]
Thanks for the quick reply!

**Comment by davecheney** [comment]
Is your freebsd system 32bit or 64 bit?

**Comment by swills** [comment]
My system is 64 bit.

**Comment by davecheney** [comment]
Hmm. Those values are known to be correct for Darwin, so I think you'll
have to copy the file and make a freebsd specific version.

**Comment by swills** [comment]
Sorry, found two test failures locally that I will look into:

--- FAIL: TestTcflush (0.00s)
        termios_bsd_test.go:16: inappropriate ioctl for device
--- FAIL: TestTcsendbreak (0.00s)
        termios_test.go:43: inappropriate ioctl for device

**Comment by davecheney** [comment]
Sure thing, no hurry.

**Comment by swills** [comment]
I'm thinking it may not be possible to make this work properly without access to the posix_openpt(4) syscall. :(

**Comment by swills** [comment]
ie, without patching the syscall package (https://golang.org/pkg/syscall/)

**Comment by davecheney** [comment]
Hmm, i'd like to fix this. There is also the golang.org/x/term repo that
we're trying to get off the ground. I'm sure it's possible to make this
work on freebsd. I'll rattle the cage on the x/term repository to see who
is in charge there.

**Comment by davecheney** [comment]
I think this has been done
